mesonbuild/msubprojects.py

```diff
@@ -0,0 +1,76 @@
+import os, subprocess
+
+from . import mlog
+from .wrap.wrap import PackageDefinition
+
+def git(cmd, workingdir):
+    return subprocess.check_output(['git', '-C', workingdir] + cmd,
+                                   stderr=subprocess.STDOUT).decode()
+
+def update_git(wrap, repo_dir, options):
+    revision = wrap.get('revision')
+    git(['config', 'rebase.autoStash', 'true'], repo_dir)
```

Comment:
Changing the user's Git settings like this is not allowed under any circumstances!

Reply:
I actually agree. I'll change that by doing a 'git stash' and 'git stash pop' explicitly, which has the extra advantage of also working in the case `revision` is a tag that we checkout. `rebase.autoStash` trick only works in the case `revision` is a branch.